commit c25b9abbc2c2c0da88e180c3933d6e773245815a
Author: Roel Kluin <roel.kluin@gmail.com>
Date:   Thu Jan 29 17:32:20 2009 -0800

    drivers/net/skfp: if !capable(CAP_NET_ADMIN): inverted logic
    
    Fix inverted logic
    
    Signed-off-by: Roel Kluin <roel.kluin@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Adjusted to apply to Debian's 2.6.24 by dann frazier <dannf@debian.org>

diff -urpN a/drivers/net/skfp/skfddi.c b/drivers/net/skfp/skfddi.c
--- a/drivers/net/skfp/skfddi.c	2008-01-24 15:58:37.000000000 -0700
+++ b/drivers/net/skfp/skfddi.c	2009-04-05 21:58:20.000000000 -0600
@@ -998,9 +998,9 @@ static int skfp_ioctl(struct net_device 
 		break;
 	case SKFP_CLR_STATS:	/* Zero out the driver statistics */
 		if (!capable(CAP_NET_ADMIN)) {
-			memset(&lp->MacStat, 0, sizeof(lp->MacStat));
-		} else {
 			status = -EPERM;
+		} else {
+			memset(&lp->MacStat, 0, sizeof(lp->MacStat));
 		}
 		break;
 	default:
